charts/px-central/templates/px-backup/service-mesh/pxcentral-istio.yaml

```diff
@@ -0,0 +1,72 @@
+{{- $istioEnabled := .Values.istio.enabled | default false }}
+{{- $deployDedicatedMonitoringSystem := .Values.pxbackup.deployDedicatedMonitoringSystem }}
+{{- if eq $istioEnabled true }}
+apiVersion: networking.istio.io/v1
+kind: Gateway
+metadata:
+  name: pxcentral-gateway
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-weight": "0"
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - hosts:
+    - '*'
+    port:
+      name: http
+      number: 80
+      protocol: HTTP
+---
+apiVersion: security.istio.io/v1
+kind: PeerAuthentication
+metadata:
+  name: pxcentral-peer-authentication
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-weight": "0"
+spec:
+  mtls:
+    mode: STRICT
+---
+apiVersion: networking.istio.io/v1
+kind: VirtualService
+metadata:
+  name: pxcentral-virtualservice
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-weight": "0"
+spec:
+  gateways:
+  - pxcentral-gateway
+  hosts:
+  - '*'
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: px-central-ui.{{ .Release.Namespace }}.svc.cluster.local
+        port:
+          number: 80
```

Comment:
If we do this, then any request that is coming to this gateway will be sent to px-central-ui service?
Now what should a user do if they want to expose the px-backup service to make API calls directly. This command alone will not suffice right?
```
kubectl expose svc px-backup --type NodePort --port 10001 --name px-backup-exposed -n px-backup
```

Reply:
Hmm, in this case , we need to create a virtual service for the px-backup api as well , will take care in the documentation